### PR TITLE
DCS-817 created Api client for user preference service

### PR DESCRIPTION
**Pull request by GurnankCheema**

### CODE REVIEW

File: @types/courtRegister/index.d.ts [hunk 1]
```diff
@@ -1,8 +1,13 @@
 declare module 'courtRegister' {
   export type CourtDtoPage = schemas['CourtDtoPage']
   export type CourtDto = schemas['CourtDto']
+  export type CourtIdDto = schemas['CourtIdDto']
 
   interface schemas {
+    /** array of court ids */
+    CourtIdDto: {
```
Comment (andrewrlee) [inline @types/courtRegister/index.d.ts:8]: this is for types returned by the court register dto - you need to create a new one of these from the api spec for the user preference file.  

something like:
```npx openapi-typescript  https://prisoner-offender-search-dev.prison.service.justice.gov.uk/v3/api-docs -output prisoner-search-api.d.ts```

File: backend/api/index.ts [hunk 2]
```diff
@@ -17,6 +18,13 @@ const courtApi = new CourtApi(
   })
 )
 
+const userCourtPreferencesApi = new UserCourtPreferencesApi(
+  new Client({
+    baseUrl: config.apis.courts.url,
```
Comment (andrewrlee) [inline backend/api/index.ts:23]: wrong url?

File: backend/config.js [hunk 3]
```diff
@@ -58,6 +58,10 @@ module.exports = {
       url: process.env.API_COURTS_ENDPOINT_URL || 'http://localhost:8084/',
       timeoutSeconds: toInt(process.env.API_COURTS_TIMEOUT_SECONDS, 30),
     },
+    userCourtPreferences: {
```
Comment (andrewrlee) [inline backend/config.js:61]: `userPreferences`

File: @types/courtRegister/index.d.ts (hunk 1 above)
Comment (GurnankCheema) [reply @types/courtRegister/index.d.ts:8]: done

File: backend/api/index.ts (hunk 2 above)
Comment (GurnankCheema) [reply backend/api/index.ts:23]: changed

File: backend/config.js (hunk 3 above)
Comment (GurnankCheema) [reply backend/config.js:61]: done

**Review by andrewrlee — APPROVED** [review]
LGTM

File: @types/userPreferences/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        | 'TOO_MANY_REQUESTS'
+        | 'UNAUTHORIZED'
+        | 'UNAVAILABLE_FOR_LEGAL_REASONS'
+        | 'UNPROCESSABLE_ENTITY'
+        | 'UNSUPPORTED_MEDIA_TYPE'
+        | 'UPGRADE_REQUIRED'
+        | 'URI_TOO_LONG'
+        | 'USE_PROXY'
+        | 'VARIANT_ALSO_NEGOTIATES'
+      view?: definitions['View']
```
Comment (andrewrlee) [inline @types/userPreferences/index.d.ts:79]: this should be `schemas['View]` (we aren't actually using it so its NBD)

**Pull request merged**